integrations/newrelic/newrelic_integration/core/service_levels.py

```diff
@@ -0,0 +1,80 @@
+from typing import Any, AsyncIterable, Tuple, Optional
+import httpx
+from port_ocean.context.ocean import ocean
+from newrelic_integration.core.query_templates.service_levels import (
+    LIST_SLOS_QUERY,
+    GET_SLI_BY_NRQL_QUERY,
+)
+from newrelic_integration.core.utils import send_graph_api_request
+from newrelic_integration.utils import (
+    render_query,
+)
+from newrelic_integration.core.paging import send_paginated_graph_api_request
+
+SLI_OBJECT = "__SLI"
+
+
+class ServiceLevelsHandler:
+    def __init__(self, http_client: httpx.AsyncClient):
+        self.http_client = http_client
+
+    async def get_service_level_indicator_value(
+        self, http_client: httpx.AsyncClient, nrql: str
+    ) -> dict[Any, Any]:
+        query = await render_query(
+            GET_SLI_BY_NRQL_QUERY,
+            nrql_query=nrql,
+            account_id=ocean.integration_config.get("new_relic_account_id"),
+        )
+        response = await send_graph_api_request(
+            http_client, query, request_type="get_service_level_indicator_value"
+        )
+        service_levels = (
+            response.get("data", {})
+            .get("actor", {})
+            .get("account", {})
+            .get("nrql", {})
+            .get("results", [])
+        )
+        if service_levels:
+            return service_levels[0]
+        return {}
+
+    async def list_service_levels(self) -> AsyncIterable[dict[str, Any]]:
+
+        async for service_level in send_paginated_graph_api_request(
+            self.http_client,
```

Comment:
if you look inside the `send_paginated_graph_api_request` it actually already fetched a batch, but returns it item by item, so to make it return faster, and maybe async gather the SLIs for multiple SLOs I would suggest either adjusting the `send_paginated_graph_api_request` or gathering a size of a batch and then bringing the `SLI` for each one, and using the `stream_async_iterators_tasks` to return the results for each one.